runtime/plugins/org.jboss.tools.runtime.core/src/org/jboss/tools/runtime/core/model/AbstractRuntimeDetectorDelegate.java

```diff
@@ -22,11 +24,32 @@
 public abstract class AbstractRuntimeDetectorDelegate implements
 		IRuntimeDetectorDelegate {
 
-	@Override
+	private boolean loggedWarning = false;
+	
+	@Override @Deprecated
 	public void initializeRuntimes(List<RuntimeDefinition> runtimeDefinitions) {
 	}
 
 	@Override
+	public boolean initializeRuntime(RuntimeDefinition runtimeDefinition) throws CoreException {
+		// Provide some migration code, for now.  This code may be removed at any time in the future. 
+		
+		// Since this is a new method, and some clients might not be implementing it, we 
+		// should redirect to the original method initializeRuntimes.
+		initializeRuntimes(Collections.singletonList(runtimeDefinition));
+		
+		// Log an error / warning indicating other handlers should update their API call
```

Comment:
> the error will be logged when calling initializeRuntime(RuntimeDefinition runtimeDefinition) 

This is intentiona.  The error will be logged when calling initializeRuntime(RuntimeDefinition runtimeDefinition)  only when a subclass of AbstractRuntimeDetectorDelegate has not yet implemented the new method.  If a subclass of AbstractRuntimeDetectorDelegate has implemented initializeRuntime(RuntimeDefinition runtimeDefinition), then it will not be using this method and no error will be logged.